cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/upgrade/operation/AwsRdsUpgradeValidatorService.java

```diff
@@ -64,6 +77,29 @@ public boolean isRdsMajorVersionSmallerThanTarget(RdsInfo rdsInfo, Version targe
         return upgradeNeeded;
     }
 
+    public void validateCustomPropertiesAdded(AuthenticatedContext authenticatedContext, DatabaseStack stack) {
+        if (stack.getDatabaseServer().isUseSslEnforcement()) {
+            LOGGER.info("Custom parameter group check is skipped as CB use its own custom group for ssl enforcement");
+        } else {
+            DescribeDBInstancesResult describeDBInstancesResult = awsRdsStatusLookupService.getDescribeDBInstancesResult(authenticatedContext, stack);
+            if (describeDBInstancesResult != null && CollectionUtils.isNotEmpty(describeDBInstancesResult.getDBInstances())) {
+                String nonDefaultParamGroupNames = describeDBInstancesResult.getDBInstances().stream()
+                        .filter(instance -> Objects.nonNull(instance.getDBParameterGroups()))
+                        .flatMap(instance -> instance.getDBParameterGroups().stream())
+                        .map(DBParameterGroupStatus::getDBParameterGroupName)
+                        .filter(groupName -> !groupName.startsWith("default."))
+                        .collect(Collectors.joining(", "));
+                if (StringUtils.isNotBlank(nonDefaultParamGroupNames)) {
+                    String message = String.format(
+                            "The following custom parameter groups are attached to the RDS instance [%s]: %s. Please remove them before RDS upgrade.",
```

Comment:
Nit, maybe : "As we could not guarantee parameter compatibility between RDS versions,   please remove them before RDS upgrade and they can be re-attached afterwards manually."